git/packet_writer.go

```diff
@@ -57,15 +58,7 @@ func (w *PacketWriter) Write(p []byte) (int, error) {
 	var n int
 
 	if p == nil {
-		// If we got an empty sequence of bytes, let's flush the data
-		// stored in the buffer, and then write the a packet termination
-		// sequence.
-
-		if _, err := w.flush(); err != nil {
-			return 0, err
-		}
-
-		if err := w.proto.writeFlush(); err != nil {
+		if err := w.Flush(); err != nil {
 			return 0, err
 		}
```

Comment:
Since we don't need `write(nil)` anymore, shouldn't we remove that feature completely?